### PR TITLE
LPD-35857: Incorrect Import Order: Category Facet Imported Before Categories, Leading to Missing Configuration

### PR DESCRIPTION
**Pull request by gustavolimav**
Ticket: https://liferay.atlassian.net/browse/LPD-35857

https://liferay.slack.com/archives/CL9RTSZ52/p1725012156819599?thread_ts=1724936693.564159&cid=CL9RTSZ52

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:sf

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/exportimport/portlet/preferences/processor/CategoryFacetExportImportPortletPreferencesProcessor.java [hunk 1]
```diff
@@ -191,6 +198,36 @@ protected Long getImportPortletPreferencesNewValue(
...
+		Element importDataRootElement =
+			portletDataContext.getImportDataRootElement();
+
+		Element referencesElement = importDataRootElement.element("references");
+
+		if (referencesElement == null) {
+			return;
+		}
+
+		List<Element> referenceElements = referencesElement.elements();
```
Comment (AlmirFe) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/exportimport/portlet/preferences/processor/CategoryFacetExportImportPortletPreferencesProcessor.java:214]: Pass  `referencesElement.elements();` directly in the for loop

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/exportimport/portlet/preferences/processor/CategoryFacetExportImportPortletPreferencesProcessor.java [hunk 2]
```diff
@@ -191,6 +198,36 @@ protected Long getImportPortletPreferencesNewValue(
...
+
+		for (Element referenceElement : referenceElements) {
+			String className = referenceElement.attributeValue("class-name");
+
+			if (!className.equals(AssetVocabulary.class.getName())) {
```
Comment (AlmirFe) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/exportimport/portlet/preferences/processor/CategoryFacetExportImportPortletPreferencesProcessor.java:219]: Check if className is null before

**Review by AlmirFe — CHANGES_REQUESTED** [review]

**Review by AlmirFe — APPROVED** [review]
It's good to make the hotfix

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Review by AlmirFe — APPROVED** [review]
LGTM! :rocket:

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:forward

**Pull request closed without merging**